test/attr/attr_availability_noasync.swift

```diff
@@ -27,6 +27,15 @@ actor IOActor {
 
 @available(SwiftStdlib 5.5, *)
 func asyncFunc() async {
+    defer {
+        defer {
+            // expected-warning@+1 {{global function 'basicNoAsync' is unavailable from asynchronous contexts; this is an error in the Swift 6 language mode}}
+            basicNoAsync()
+        }
+        // expected-warning@+1 {{global function 'basicNoAsync' is unavailable from asynchronous contexts; this is an error in the Swift 6 language mode}}
+        basicNoAsync()
+    }
```

Comment:
Some additional test cases that would be useful to have to ensure the walk up the context hierarchy remains correct would be to test a `defer` inside:
- either a sync or an async closure or local function
- inside either a sync or an async function.
I think there are 8 total combinations there, though it may be that some of them are redundant.